blueprints/api.guidelines.md

```diff
@@ -0,0 +1,535 @@
+contact some other service on which the endpoint's processes depend. It MAY be
+used (with caution) by plugin developers, but SHOULD NOT be used by authors of
+proper API endpoints, as that divulges information about failing connected
+systems and potentially gives an attacker information about Traffic Control's
+weak points. API endpoint authors should instead use `500 Internal Server
+Error`.
+
+### Documentation
+All endpoints must be properly documented. For guidelines for writing API
+documentation, refer to
+[that section of the official ATC documentation](https://traffic-control-cdn.readthedocs.io/en/latest/development/documentation_guidelines.html#documenting-api-routes).
+
+### Passing Request Data
+Request data may be passed in the request body or as a
+`application/x-www-form-urlencoded`-encoded query string in the request URI, or
+as a part of the request path. Request data MUST NOT be passed through a portion
+of the request path unless it uniquely identifies a resource with which the
+client may interact. For example, `/foos/{{ID}}}` is an acceptable path for
+dealing with the particular "Foo" object that has some identifier `ID`, but
+`logs/{{Number of Days}}/days` is unacceptable because reasonable default
+behavior can be provided if no number of days is given in the query string
+parameters, and that doesn't help uniquely identify a resource. Request path
+parameters should use double "curly-braces" (<kbd>{</kbd> and <kbd>}</kbd>) to
+call out variable components of the request path in documentation and
+references. Request path parameters MUST NOT be used for data that is optional
+to the request (somewhat obviously). Note that all endpoints dealing with the
+manipulation of "Traffic Control Objects"  MUST support the GET HTTP request
+method.
+
+When accepting data in the request body of requests, the endpoint MUST properly
+document the object representations (properties and their types) it accepts and
+MUST reject semantically invalid data with a `400 Bad Request`.
+
+The decision to pass data in the request body or query string is mainly up to
+the author, but some helpful tips:
+
+- GET and DELETE requests do not typically provide request bodies.
+- Query parameters should nearly always be optional. If data is required by an
+endpoint, consider requiring it in the request body. If the data identifies a
+resource, it ought to be a path parameter.
+- Request body data often represents objects that are being created or updated.
+If an object is being created or updated, it ought to be defined in the request
+body, and if any additional data is (possibly optionally) required then it ought
+to be passed in the query string to separate it from the object definition.
+- The following query parameters are reserved for special use by Traffic Ops
+endpoint handlers, and may not be used for any purpose other than their
+prescribed functions.
+
+	- `limit`
+	- `newerThan`
+	- `offset`
+	- `olderThan`
+	- `orderby`
```

Comment:
should this be `orderBy`?

Reply:
nope. Our current API uses "orderby," unfortunately. Changing that would be a breaking change unless they both worked - and I don't think anyone wants to bother with that. I know I don't. So we'll save consistent casing for v3 (or even v4 maybe, at this point).